R/latent_individual.R

```diff
@@ -138,94 +194,55 @@ epidist_formula.epidist_latent_individual <- function(data, delay_central = ~ 1,
       )
     )
   }
-  delay_equation <- paste0(
-    "delay_central | vreal(obs_t, pwindow_upr, swindow_upr)",
-    paste(delay_central, collapse = " ")
-  )
-  sigma_equation <- paste0("sigma", paste(sigma, collapse = " "))
-  form <- brms::bf(as.formula(delay_equation), as.formula(sigma_equation))
-  return(form)
-}
-
-#' @importFrom rstan lookup
-#' @method epidist_family epidist_latent_individual
-#' @family latent_individual
-#' @export
-epidist_family.epidist_latent_individual <- function(data, family = "lognormal",
-                                                     ...) {
-  epidist_validate(data)
-  checkmate::assert_string(family)
-
-  pdf_lookup <- rstan::lookup("pdf")
-  valid_pdfs <- gsub("_lpdf", "", pdf_lookup$StanFunction)
-  if (!family %in% valid_pdfs) {
-    cli::cli_warn(
-      "The provided family {.code family} does not correspond to a valid LPDF
-      function available in rstan. (It is possible [but unlikely] that there is
-      such an LPDF in Stan available via cmdstanr, as rstan is behind Stan.)"
-    )
-  }
-
-  brms::custom_family(
-    paste0("latent_", family),
-    dpars = c("mu", "sigma"),
-    links = c("identity", "log"),
-    lb = c(NA, 0),
-    ub = c(NA, NA),
-    type = "real",
-    vars = c("pwindow", "swindow", "vreal1"),
-    loop = FALSE
+  mu_index <- which(dpars == "mu")
+  formula[[mu_index]] <- update(
```

Comment:
Current `update` simplifies the formula e.g. going from `1 + sex` to just `sex` (as the intercept is implicitly there). I could write a version of `update` which doesn't do that (sadly `simplify = TRUE` is hard coded into it atm). It could involve adding a function to this package though